staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -391,6 +416,7 @@ func (e *Store) Create(ctx context.Context, obj runtime.Object, createValidation
 			return nil, err
 		}
 	}
+	success = true // used in the defer'ed cleanup
```

Comment:
So this was not clear to me.  Why do we allow After* and Decorator to return an error at all?  If they fail, it seems like we hit a pretty awful path - operation failed but object exists?

Given how little those are used - can we just make them not return an error?

I can flip the sequence so that cleanup happens before these and just pretend this weirdness doesn't exist, of course....

Reply:
> Why do we allow After* and Decorator to return an error at all? If they fail, it seems like we hit a pretty awful path - operation failed but object exists?

Agree. "Why do we even have this lever?"

> Given how little those are used - can we just make them not return an error?

That would be my inclination. @smarterclayton?

Reply:
ICYMI - I added commits that implement this - nobody (at least in-tree) uses those hooks to return errors.